### PR TITLE
Allow optional group and more than one mandatory argument in nospell commands

### PR DESCRIPTION
**Pull request by tornaria**
When a command is in `g:vimtex_syntax_nospell_commands`, its argument is excluded from spell checking; after the first patch this also works for commands with an optional argument.

After the second patch, a command in `g:vimtex_syntax_nospell_commands` can have more than one mandatory argument; spell check is disabled for all of them.

I use this for custom commands with optional arguments; also for some standard commands like `\numberwithin` (2 mandatory arguments).

### CODE REVIEW

File: autoload/vimtex/syntax/core.vim [hunk 1]
```diff
@@ -758,6 +759,8 @@ function! vimtex#syntax#core#init_highlights() abort " {{{1
   highlight def link texNewthmArgName      texArg
   highlight def link texNewthmOptCounter   texOpt
   highlight def link texNewthmOptNumberby  texOpt
+  highlight def link texNoSpellOpt         texOpt
+  highlight def link texNoSpellArg         texArg
```
Comment (lervag) [inline autoload/vimtex/syntax/core.vim:763]: I'm not sure if we really want to add the default highlight for `texNoSpellArg` here. Users can add that if they want, but I thought it OK to just keep it transparent.

**Review by lervag — CHANGES_REQUESTED** [review]
Great, thanks. I don't see any reason not to accept this, except my minor comment.

File: autoload/vimtex/syntax/core.vim (hunk 1 above)
Comment (tornaria) [reply autoload/vimtex/syntax/core.vim]: You are right, done.

**Comment by lervag** [comment]
Thanks!

**Pull request closed without merging**